### PR TITLE
Remove all the unwraps, and fix some logging

### PR DESCRIPTION
**Pull request by nyospe**
Removing potential for panic! and improving some logging. Also cleaning up some state that should not have been retained.

### CODE REVIEW

File: src/builder_state.rs [hunk 1]
```diff
@@ -692,7 +705,7 @@ impl<TYPES: NodeType> BuilderProgress<TYPES> for BuilderState<TYPES> {
                 requested_view_number,
             ));
 
-            let encoded_txns: Vec<u8> = payload.encode().unwrap().to_vec();
+            let encoded_txns: Vec<u8> = payload.encode().ok()?.to_vec();
```
Comment (jbearer) [inline src/builder_state.rs:708]: We should log something if we can't encode the payload

File: src/builder_state.rs [hunk 2]
```diff
@@ -971,12 +995,8 @@ impl<TYPES: NodeType> BuilderState<TYPES> {
...
             qc_receiver: self.qc_receiver.clone(),
             req_receiver,
-            da_proposal_payload_commit_to_da_proposal: self
-                .da_proposal_payload_commit_to_da_proposal
-                .clone(),
-            quorum_proposal_payload_commit_to_quorum_proposal: self
-                .quorum_proposal_payload_commit_to_quorum_proposal
-                .clone(),
+            da_proposal_payload_commit_to_da_proposal: HashMap::new(),
+            quorum_proposal_payload_commit_to_quorum_proposal: HashMap::new(),
```
Comment (jbearer) [inline src/builder_state.rs:999]: Why this change?

Comment (nyospe) [reply src/builder_state.rs:999]: Potential memory leak if we're seeing multiple competing proposals (e.g. there's a malicious leader, so we shouldn't ever hit that, but still) that compounds as the outstanding number of possible parent proposals.

File: src/builder_state.rs (hunk 1 above)
Comment (nyospe) [reply src/builder_state.rs:708]: I think that's a can't fail for the actual Payload, but, yeah, the trait makes no such guarantee.

Comment (nyospe) [reply src/builder_state.rs]: Fixed in [f1322c5](https://github.com/EspressoSystems/hotshot-builder-core/pull/125/commits/f1322c50ba053956be034172b621ab0ab4ce3704)

**Review by jbearer — APPROVED** [review]
